mm2src/coins/rpc_command/get_current_mtp.rs

```diff
@@ -0,0 +1,78 @@
+use common::{HttpStatusCode, StatusCode};
+use derive_more::Display;
+use mm2_core::mm_ctx::MmArc;
+use mm2_err_handle::prelude::MmError;
+
+use crate::{lp_coinfind_or_err,
+            utxo::{rpc_clients::UtxoRpcError, UtxoCommonOps},
+            CoinFindError, MmCoinEnum};
+
+pub type GetCurrentMtpRpcResult<T> = Result<T, MmError<GetCurrentMtpError>>;
+
+#[derive(Deserialize)]
+pub struct GetCurrentMtpRequest {
+    coin: String,
+}
+
+#[derive(Serialize)]
+pub struct GetCurrentMtpResponse {
+    mtp: u32,
+}
+
+#[derive(Serialize, Display, SerializeErrorType)]
+#[serde(tag = "error_type", content = "error_data")]
+pub enum GetCurrentMtpError {
+    NoSuchCoin(String),
+    NotSupportedCoin(String),
+    UtxoRpcError(String),
+}
+
+impl HttpStatusCode for GetCurrentMtpError {
+    fn status_code(&self) -> StatusCode {
+        match self {
+            GetCurrentMtpError::NoSuchCoin(_)
+            | GetCurrentMtpError::UtxoRpcError(_)
+            | GetCurrentMtpError::NotSupportedCoin(_) => StatusCode::BAD_REQUEST,
+        }
+    }
+}
+
+impl From<UtxoRpcError> for GetCurrentMtpError {
+    fn from(err: UtxoRpcError) -> Self { Self::UtxoRpcError(err.to_string()) }
+}
+
+impl From<CoinFindError> for GetCurrentMtpError {
+    fn from(err: CoinFindError) -> Self {
+        match err {
+            CoinFindError::NoSuchCoin { coin } => Self::NoSuchCoin(format!("No such coin: {}", coin)),
+        }
```

Comment:
`CoinFindError::NoSuchCoin` is already formated to `"No such coin: {}", coin)`https://github.com/KomodoPlatform/atomicDEX-API/blob/84808f3b063b5068a44e656c6924fc0dae9fc357/mm2src/coins/lp_coins.rs#L2453-L2457